### PR TITLE
BLD: update minimum Python, NumPy, Cython, Pybind11 versions

### PR DESCRIPTION
**Pull request by rgommers**
Now that 1.4.0 is out the door, it's time for another set of version bumps. Dropping Python 3.5 and NumPy 1.13, and updating Cython and Pybind11 to latest (there's a necessary Python 3.8 in both, and useful fixes for other issues as well).

This PR is also useful preparation for a Python 2.7 cleanup that is okay to start doing on Jan 1st. Given that people like deleting stuff, I'm sure that PRs will arrive soon-ish:)

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -112,10 +106,10 @@ jobs:
       choco install -y mingw --force --version=6.4.0
     displayName: 'Install 64-bit mingw for 64-bit builds'
     condition: eq(variables['BITS'], 64)
-  - script: python -m pip install numpy cython==0.29.13 pybind11 pytest pytest-timeout pytest-xdist pytest-env pytest-cov Pillow mpmath
+  - script: python -m pip install numpy cython==0.29.14 pybind11 pytest pytest-timeout pytest-xdist pytest-env pytest-cov Pillow mpmath
     displayName: 'Install dependencies'
   - powershell: |
-      If ($(PYTHON_VERSION) -ne '3.5') {
+      If ($(PYTHON_VERSION) -ne '3.6') {
```
Comment (tylerjereddy) [inline azure-pipelines.yml]: With 3.5 out of the matrix we shouldn't need this conditional, but I'll just clean that up once the final matplotlib 3.2.0 comes out, presumably after the winter break.

File: .travis.yml [hunk 2]
```diff
@@ -52,18 +52,18 @@ matrix:
...
         - TESTMODE=fast
         - NUMPYSPEC="--upgrade numpy"
       addons:
         apt:
           packages:
             - *common_packages
-            - python3.5-dbg
-            - python3.5-dev
+            - python3.7-dbg
+            - python3.7-dev
```
Comment (tylerjereddy) [inline .travis.yml]: Looks like Travis can't find these package names---maybe they are slightly different now or we need a higher version base image of the distro?

**Review by tylerjereddy — COMMENTED** [review]
Thanks Ralf --- looks like the Travis debug build will need a small tweak somewhere. Beyond that, the other CI failures look unrelated/already known elsewhere.

**Comment by charris** [comment]
NumPy uses bionic to get the 3.6-dbg packages. The 3.7-dbg will be in the next LTS in 2020.
```
    - python: 3.6
      dist: bionic
      env: USE_DEBUG=1
      addons:
        apt:
          packages:
            - *common_packages
            - cython3-dbg
            - python3-dbg
            - python3-dev
            - python3-setuptools
```

**Comment by tylerjereddy** [comment]
May have to pester me a bit if we want to get another `1.2.x` LTS release out the door as well? I know I have a backport PR open for that and it was looking a little annoying & then the 1.3.x and 1.4.x series took precedence near the end of the year.

**Comment by rgommers** [comment]
> May have to pester me a bit if we want to get another `1.2.x` LTS release out the door as well? I know I have a backport PR open for that and it was looking a little annoying & then the 1.3.x and 1.4.x series took precedence near the end of the year.

Thanks Tyler, I'll comment on gh-10758

**Comment by rgommers** [comment]
okay this is good to go now

**Comment by tylerjereddy** [comment]
Merged, thanks @rgommers

File: .travis.yml [hunk 3]
```diff
@@ -186,7 +186,7 @@ before_install:
   - |
     if [ -n "${USE_DEBUG}" ]; then
         # see gh-10676; need to pin pytest version with debug
-        # Python 3.5
+        # Python 3.7
```
Comment (tylerjereddy) [inline .travis.yml]: I think this comment should indicate Python 3.6 after the revisions, but wasn't even worth flushing the CI for such a small thing IMO. Catch in the next cleanup.

Comment (rgommers) [reply .travis.yml]: Thanks, good call Tyler - we should get in this habit more, merge first and flag minor things for follow-up. I have another CI PR to make today, so will take it along there.

**Pull request merged**